src/dialog-user/services/dialogData.ts

```diff
@@ -95,14 +103,14 @@ export default class DialogDataService {
       defaultValue = data.values[firstOption][fieldValue];
     }
 
-    if (data.type === 'DialogFieldDateControl' || data.type === 'DialogFieldDateTimeControl') {
-      defaultValue = data.values ? new Date(data.values) : new Date();
-    }
-
     if (data.default_value) {
       defaultValue = data.default_value;
     }
 
+    if (data.type === 'DialogFieldDateControl' || data.type === 'DialogFieldDateTimeControl') {
+      defaultValue = data.dateField ? new Date(data.dateField) : new Date();
```

Comment:
I think my only concern here is if `data.dateField` is going to ever be populated for `DialogFieldDateTimeControl`, because up on line 46, we are only setting `field.dateField` for `DialogFieldDateControl`. Is that a problem?

Reply:
> because up on line 46, we are only setting field.dateField for DialogFieldDateControl

@eclarizio we're setting the `data.dateField` and `data.timeField` for `DialogFieldDateTimeControl` right above the `DialogFieldDateControl`, on line L38-L44:

```javascript
 38     if (field.type === 'DialogFieldDateTimeControl') {
 39       if (_.isNull(field.values) || _.isUndefined(field.values)) {
 40         field.dateField = field.timeField = new Date();
 41       } else {
 42         field.dateField = field.timeField = new Date(data.values);
 43       }
 44     }
```

Reply:
I'm still confused about this change here (moving defaultValue=... *under* the other if).

Previously, `default_value` coming from the server would get used.
Now, only `data.dateField` will ever be used for date/time values.

So... 
a) why not `data.timeField` also?
b) should we really never use `default_value` from the server when dealing with date&time?

Reply:
> why not data.timeField also?

the `timeField` is only related to `DialogFieldDateTimeControl`, there's no time field in `DialogFieldDateControl`

Reply:
> should we really never use default_value from the server when dealing with date&time?

we're using the `default_value` on the line 46 that @eclarizio has mentioned

Reply:
OK, so `default_value` is now read only in  `setupField` and we're using `dateField` only in `setDefaultValue`.

But, that still doesn't explain why `timeField` gets ignored when in a `if (data.type === 'DialogFieldDateControl' || data.type === 'DialogFieldDateTimeControl')` condition ;).

Reply:
@himdel you're right. The `(data.type === 'DialogFieldDateControl' || data.type === 'DialogFieldDateTimeControl')` condition should be split. I'll update the PR.

Reply:
ah, LGTM then, thanks :)